src/mono/wasm/README.md

```diff
@@ -1,25 +1,42 @@
-# Building
+# Build WebAssembly
 
-This depends on `emsdk` to be installed.
+If you haven't already done so, please read [this document](../../../docs/workflow/README.md#Build_Requirements) to understand the build requirements for your operating system. If you are specifically interested in building libraries for WebAssembly, read [Libraries WebAssembly](../../../docs/workflow/building/libraries/webassembly-instructions.md).
 
-## emsdk on macOS
+## Installing emsdk
+
+The **correct version** of Emscripten SDK (emsdk) needs to be installed. Version number is saved in [emscripten-version.txt](./emscripten-version.txt).
+
+### macOS/Linux
 
 * You can run `make provision-wasm`, which will install it to `$reporoot/src/mono/wasm/emsdk` .
 Note: Irrespective of `$(EMSDK_PATH)`'s value, `provision-wasm` will always install into `$reporoot/src/mono/wasm/emsdk`.
 
-`EMSDK_PATH` is set to `$reporoot/src/mono/wasm/emsdk` by default, by the Makefile.
-
-Note: `EMSDK_PATH` is set by default in `src/mono/wasm/Makefile`, so building targets from that will have it set. But you might need to set it manually if
-you are directly using the `dotnet build`, or `build.sh`.
+Note: `EMSDK_PATH` is set by default in `src/mono/wasm/Makefile`, so building targets from that will have it set. But you might need to set it manually if you are directly using the `dotnet build`, or `build.sh` by `source ./emsdk_env.sh`.
```

Comment:
IMHO because using CMake is not the way how CI works and so it's always lagging behind and breaks easily